### ISSUE TITLE
Feature request: suggest first completion "recentlyUsedByPrefix"

### ISSUE THREAD

**Issue by tlannaax** [neoclide/coc.nvim#2575]
When editing files containing large parts of normal text (like latex, markdown, documentation), it is often that several variants of a word (verb, noun, single form, plural form, adj., ...) appear in the keyword completion list.
Like introduce/introduction(s)/introductory,  mathematics/mathmatician/mathematical, they have most charaters in common.
In most cases the word in the first place of completion is not the one I want to input.
Sometimes fuzzy match can help to resolve the amibiguity, usually by inputing the ending charater. 
But sometimes not, for example when the ending character also appear in the middle commen part, and when it doesn't, you just wasted more time and have to use <C-N> to select what you want.
In coding, such situation may be avoided by carefully designed naming convention. However, in natural languages it is inevitabl. 

To this end, I found the feature "editor.suggestSelection": "recentlyUsedByPrefix" available on VSCode is a life saver.
Basically, it remembers the choices of completion for each prefix you have inputted, and at the next appearance of the same prefix, the remembered choice will be put in the first place of the completion menu.
This way you can use the prefix you like to quickly complete different variants of usual words, especially those with irregular forms of variants.
